main.js

```diff
@@ -0,0 +1,85 @@
+function allowDrop(ev) {
+  ev.preventDefault();
+}
+
+function drag(ev, e) {
+  ev.dataTransfer.setData("id", e.id);
+  ev.dataTransfer.setData("sourceId", e.parentNode.id);
+}
+
+function drop(ev) {
+  let id = ev.dataTransfer.getData("id");
+  let sourceId = ev.dataTransfer.getData("sourceId");
+  let targetId = ev.target.id;  
+  if (id != '' && sourceId == 'marbleBox') {
+    let guessDiv = document.getElementById(targetId);
+    let guessCount = guessDiv.childNodes.length;
+    if (guessCount < 4) {
+      ev.preventDefault();
+      let marble = document.getElementById(id).cloneNode(true);
+      ev.target.appendChild(marble);
+      if (guessCount == 3) {
+        checkUserGuess(guessDiv, targetId)
+      }
+    }
+  }
+}
+
+let cpuGuess = [];
+
+let generateRandomGuess = () => {
+  let i = 1
+  while (i <= 4) {
+    let b = 5, a = 1;
+    let num = Math.random() * (b - a) + a;
+    let numFloor = Math.floor(num)
+    cpuGuess.push(numFloor)
+    i++
+  }
+  console.log(cpuGuess)
+}
+
+generateRandomGuess();
+
+let checkUserGuess = (e, targetId) => {
```

Comment:
This code does not correctly give the second part of the hint, the one stored in the "color" attribute